editor/import/3d/resource_importer_obj.cpp

```diff
@@ -445,6 +445,42 @@ static Error _parse_obj(const String &p_path, List<Ref<ImporterMesh>> &r_meshes,
 				}
 
 				mesh->add_surface(Mesh::PRIMITIVE_TRIANGLES, array, TypedArray<Array>(), Dictionary(), material, name, mesh_flags);
+
+				if (p_generate_lightmap_uv2) {
+					Vector<uint8_t> lightmap_cache;
+					mesh->lightmap_unwrap_cached(Transform3D(), p_generate_lightmap_uv2_texel_size, p_src_lightmap_cache, lightmap_cache);
+
+					if (!lightmap_cache.is_empty()) {
+						if (r_lightmap_caches.is_empty()) {
+							r_lightmap_caches.push_back(lightmap_cache);
+						} else {
+							// MD5 is stored at the beginning of the cache data.
+							const String new_md5 = String::md5(lightmap_cache.ptr());
+
+							for (int i = 0; i < r_lightmap_caches.size(); i++) {
+								const String md5 = String::md5(r_lightmap_caches[i].ptr());
+								if (new_md5 < md5) {
+									r_lightmap_caches.insert(i, lightmap_cache);
+									break;
+								}
+
+								if (new_md5 == md5) {
+									break;
+								}
+							}
+						}
+					}
+				}
+
+				if (p_generate_lods) {
+					// Use normal merge/split angles that match the defaults used for 3D scene importing.
+					mesh->generate_lods(60.0f, 25.0f, {});
+				}
+
+				if (p_generate_shadow_mesh) {
+					mesh->create_shadow_mesh();
+				}
```

Comment:
Note: if you perform these steps *before* lightmap UV2 generation, the resulting mesh will be broken. It'll display with the wrong transform and no material (pure black color). I have no idea why, and putting up a MRP will likely require exposing additional ImporterMesh methods (such as `create_shadow_mesh()`).